lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -145,6 +157,15 @@ bool ArrayAllocation::mapInitializationStores() {
 
     ElementValueMap[IndexVal.getZExtValue()] = SI->getSrc();
   }
+
+  // Create a vector from the map as well, so append(contentsOf: [A]) calls are
+  // easier to replace
+  auto elementCount = ElementValueMap.size();
+  for (size_t i = 0; i < elementCount; ++i) {
+    assert(ElementValueMap.count(i) && "Elements should be contiguous");
```

Comment:
You should rather bail than assert if there is no defined value for a certain index.
I'm not sure if the compiler currently will create such a SIL where only parts of the elements are initialized by stores, it would definitely be valid SIL.